### PR TITLE
Fix/default python during nox test

### PR DESCRIPTION
**Pull request by spazm**
* [x] I have added an entry to `docs/changelog.md`

## Test plan

Tested by running
```
# command(s) to exercise these changes

% nox -s docs
# view changes in docs/docs.md
% grep overridden docs/docs.md | tail -1
/usr/bin/python and can be overridden by setting the environment

# compare with output when `PIPX__DOX_DEFAULT_PYTHON` is not set.
% pipx install --help |grep overridden |tail -1
/homd/spazm/src/github/spazm/pipx/venv/bin/python and can be overridden by setting the environment
```

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -77,6 +77,7 @@ def lint(session):
 @nox.session(python="3.8")
 def docs(session):
     session.install(*doc_dependencies)
+    session.env["PIPX__DOC_DEFAULT_PYTHON"] = "/usr/bin/python"
```
Comment (itsayellow) [inline noxfile.py:80]: How about instead of 
`/usr/bin/python`
we use the text
`typically the python used to execute pipx`
?

Comment (spazm) [reply noxfile.py]: This is a great idea, it looks much better when rendered.

I've made the change and also adjusted the line break for better fill.

**Review by itsayellow — CHANGES_REQUESTED** [review]
Looks good, just need to pass Lint (re-run black on `noxfile.py`)

**Comment by spazm** [comment]
Hiya pipx team.  

Thanks for a lovely project.  If you enjoy and/or approve this PR and #538 , I'd love if you tagged them both with `hacktoberfest-accepted` for hacktoberfest2020.

Happy Code-o-ween!
@spazm

**Review by itsayellow — APPROVED** [review]

**Pull request merged**